src/parser/monk/mistweaver/modules/spells/essences/WayOfTheCrane.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+import StatisticBox from 'interface/others/StatisticBox';
+import ItemHealingDone from 'interface/others/ItemHealingDone';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber } from 'common/format';
+
+
+
+/**
+ * Simple info graphic that shows what spell did what healing during wotc window, the hps of each spell, and hptc
+ */
+
+class WayOfTheCrane extends Analyzer {
+
+  customMap = null;
+  damageSpell = "";
+  lastTimeStamp = 0;
+  inWotc = false;
+  wotcTime = 0;
+
+  constructor(...args) {
+    super(...args);
+    const hasEssence = this.selectedCombatant.hasEssence(SPELLS.CONFLICT.traitId);
+    if(!hasEssence){
+        return;
+    }
+    this.active = this.selectedCombatant.hasMajor(SPELLS.CONFLICT.traitId);
+    if (!this.active) {
+      return;
+    }
+    this.customMap = new Map();
+  }
+
+  on_byPlayer_damage(event) {
+    if(!this.selectedCombatant.hasBuff(SPELLS.WAY_OF_THE_CRANE.id)){
+      return;
+    }
+    this.damageSpell = event.ability.name;
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.WAY_OF_THE_CRANE_HEAL.id || spellId === SPELLS.WAY_OF_THE_CRANE_HEAL_HONOR.id){
+      if(this.customMap.get(this.damageSpell)===undefined){
+        this.customMap.set(this.damageSpell, [1, event.amount || 0, event.overheal || 0]);
+      }else{
+        let array = this.customMap.get(this.damageSpell);
```

Comment:
instead of using an array for this, you should use an object. It is much easier to read:

```js
this._wotcHealing[this.damageSpell].heal += event.amount;
this._wotcHealing[this.damageSpell].overheal += event.overheal || 0;
```

than what you currently have